src/transactions/TransactionFrame.cpp

```diff
@@ -1772,8 +1792,19 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
             // case
             if (success)
             {
-                app.checkOnOperationApply(op->getOperation(), opResult,
-                                          ltxOp.getDelta());
+                auto delta = ltxOp.getDelta();
+
+                if (protocolVersionIsBefore(ledgerVersion,
+                                            ProtocolVersion::V_8) &&
+                    app.getConfig().EMIT_CLASSIC_EVENTS &&
+                    app.getConfig().BACKFILL_STELLAR_ASSET_EVENTS)
+                {
+                    reconcileEvents(getSourceID(), op->getOperation(), opResult,
+                                    delta, opEventManager);
```

Comment:
The `reconcileEvents` should probably just be a member function on the `opEventManager`. 
It already had the `OperationFrame` (before you removed it), the sourceID (opframe->parentTxFrame->getSourceID) the protocol number (so the condition checking could be done inside).

Reply:
I wanted to keep it separate because it's only relevant pre-protocol 8, so in most cases, there's no reason for a developer to look at this code.

Reply:
Added a comment